apps/web/src/services/dashboards/dashboard-customize/modules/dashboard-manage-variable-overlay/modules/DashboardManageVariableOptionsField.vue

```diff
@@ -84,38 +84,38 @@ import type { OptionItem } from '@/services/dashboards/dashboard-customize/modul
 interface Props {
     options: OptionItem[];
     optionsType: 'MANUAL' | 'DATA_SOURCE';
-    isManualOptionsType: boolean;
 }
 interface EmitFn {
     (e: string, value: string): void;
     (e: 'save-click', value: string): void;
     (e: 'cancel-click'): void;
+    (e: 'update-options-invalid', value: boolean): void;
 }
 const props = defineProps<Props>();
 const emit = defineEmits<EmitFn>();
 
 const state = reactive({
     proxyOptions: useProxyValue<OptionItem[]>('options', props, emit),
     proxyOptionsType: useProxyValue('optionsType', props, emit),
-    isManualOption: true,
+    manualOptionValidations: computed(() => {
+        const keys = state.proxyOptions.map((option) => option.key);
+        const labels = state.proxyOptions.map((option) => option.label);
+        return state.proxyOptions.map((option) => {
+            const validationResult = { invalid: false };
+            const isDuplicated = keys.filter((k) => k === option.key).length > 1 || labels.filter((l) => l === option.label).length > 1;
+
+            if (option.key === '' || option.label === '') validationResult.invalid = true;
+            else if (isDuplicated) validationResult.invalid = true;
+
+            return validationResult;
+        });
+    }),
```

Comment:
suggestion: If there is only `invalid` in the object, it seems to be okay to use an **array**.
```typescript
let invalidResults: boolean[] = [];
```

Reply:
I wrote that for scalability such as message, but I think so too. Thank you!!!